javatests/arcs/android/devtools/JsonEncodingsTest.kt

```diff
@@ -0,0 +1,137 @@
+package arcs.android.devtools
+
+import arcs.core.crdt.CrdtEntity
+import arcs.core.crdt.VersionMap
+import arcs.core.data.FieldType
+import arcs.core.data.RawEntity
+import arcs.core.data.util.ReferencableList
+import arcs.core.data.util.toReferencable
+import arcs.core.storage.Reference
+import arcs.core.storage.testutil.DummyStorageKey
+import arcs.core.util.JsonValue.JsonArray
+import arcs.core.util.JsonValue.JsonBoolean
+import arcs.core.util.JsonValue.JsonNumber
+import arcs.core.util.JsonValue.JsonObject
+import arcs.core.util.JsonValue.JsonString
+import com.google.common.truth.Truth.assertThat
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.junit.runners.JUnit4
+
+@RunWith(JUnit4::class)
+class JsonEncodingsTest {
+
+  @Test
+  fun versionMap_toJson() {
+    assertThat(
+      VersionMap(
+        mapOf(
+          "foo" to 1,
+          "bar" to 42
+        )
+      ).toJson()
+    ).isEqualTo(
+      JsonObject(
+        "foo" to JsonNumber(1.toDouble()),
+        "bar" to JsonNumber(42.toDouble())
+      )
+    )
+  }
+
+  @Test
+  fun referencable_primitives_toJson() {
+    assertThat("foo".toReferencable().toJson()).isEqualTo(JsonString("foo"))
+    assertThat(true.toReferencable().toJson()).isEqualTo(JsonBoolean(true))
+    assertThat(3.25.toReferencable().toJson()).isEqualTo(JsonNumber(3.25))
+    assertThat(3.25.toFloat().toReferencable().toJson()).isEqualTo(JsonNumber(3.25))
+    assertThat(10.toByte().toReferencable().toJson()).isEqualTo(JsonNumber(10.0))
+    assertThat(10.toReferencable().toJson()).isEqualTo(JsonNumber(10.0))
+    assertThat(10L.toReferencable().toJson()).isEqualTo(JsonNumber(10.0))
+  }
+
+  @Test
+  fun referencable_rawEntity_toJson() {
+    assertThat(
+      RawEntity(
+        id = "a:b:c",
+        singletons = mapOf(
+          "nickname" to "Ollie".toReferencable(),
+          "isCool" to true.toReferencable()
+        ),
+        collections = mapOf(
+          "names" to setOf("Brian".toReferencable(), "John".toReferencable())
+        ),
```

Comment:
Is it worth testing that singletons/collections within an Entity with values of type `Reference` can be Jsonified as well as primitives?  Also maybe ordered-lists?

Reply:
Added.